api/deploy.go

```diff
@@ -88,6 +88,9 @@ func deploy(w http.ResponseWriter, r *http.Request, t auth.Token) (err error) {
 	if err != nil {
 		return &tsuruErrors.HTTP{Code: http.StatusNotFound, Message: err.Error()}
 	}
+	if (instance.GetPlatform() == "") && (origin != "image") {
+		return &tsuruErrors.HTTP{Code: http.StatusUnauthorized, Message: "can't deploy app without platform, if it's not an image"}
```

Comment:
I think this logic should be based on `DeployKind` instead of origin. The origin is only informative and can be spoofed.